### ISSUE TITLE
Error "You don't have an extension of ..." with debugger extension

### ISSUE THREAD

**Issue by Sen-real** [microsoft/vscode#123881]
Hi experts,

With VS Code 1.56, we get a new problem with our debugger extension:
https://github.com/Autodesk-AutoCAD/AutoLispExt/issues/143

It seems common, as I found a similar problem here:
https://stackoverflow.com/questions/67430297/vscode-you-dont-have-an-extension-for-debugging-html-error

With our debugger extension activated, when starting to debug wth F5, an error dialog of "You don't have an extension of ..." shows up. It suggests to find our extension in the Marketplace.

But as suggested in the page above, the error is gone after we manually added a configuration file for our debugger extension.

Is this a new requirement that a user of a debugger extension must first author a configuration file before starting to debug? or is there some changes in VS Code debugger API that we should apply?

**Comment by egamma** [comment]
Duplicate of https://github.com/microsoft/vscode/issues/123106 will be fixed in 1.56.2

**Comment by nigma456** [comment]
> Duplicate of #123106 will be fixed in 1.56.2

This issue is not fixed in latest 1.56.2 and even 1.57.

**Comment by Sen-real** [comment]
@egamma @isidorn 
It's not fixed with 1.56.2. Here's a video **made with 1.56.2**:
https://user-images.githubusercontent.com/57521938/118391492-2430d980-b667-11eb-90d5-5ea7e599750e.mov

**Comment by isidorn** [comment]
@Sen-real that is for the `.lsp` file
In order to fix that your debugger extensino has to say it is interested in `.lsp` language files. You can do that by listing this in your package.json. Here's an example how mock debug does it https://github.com/microsoft/vscode-mock-debug/blob/main/package.json#L135

**Comment by Sen-real** [comment]
> @Sen-real you are showing a different issue, that is for the `.lsp` file
> In order to fix that your debugger extensino has to say it is interested in `.lsp` language files. You can do that by listing this in your package.json. Here's an example how mock debug does it https://github.com/microsoft/vscode-mock-debug/blob/main/package.json#L135

@isidorn:
Thanks for looking at this issue. We have two languages listed in package.json:
https://github.com/Autodesk-AutoCAD/AutoLispExt/blob/main/package.json#L373, and 
https://github.com/Autodesk-AutoCAD/AutoLispExt/blob/main/package.json#L385
Should this be updated somehow?
![image](https://user-images.githubusercontent.com/57521938/118447900-e04fda00-b723-11eb-834c-c8ef9d582f5f.png)

**Comment by isidorn** [comment]
@Sen-real I think you overdid it there a bit. The languages array inside the debuggers is an array of strings, not objects. So just try
`"languages": ["autolist", "autolistdcl"]`

And the full language details should be contributed I think in another package.json place, not inside `debuggers`

**Comment by Sen-real** [comment]
Thanks! let me try.

**Comment by Sen-real** [comment]
@isidorn
I just tried the change you suggested. It works well with a ".lsp" file opened in VS Code.
But, if I open NO folder or file, when I click "Run -> Start Debugging", there's still one difference:
![image](https://user-images.githubusercontent.com/57521938/118466586-92dc6880-b735-11eb-9f25-3e381936d0a1.png)
Is the new result with 1.56.2 by design?

**Comment by isidorn** [comment]
Great thanks.
That one should be fixed in insiders and should be available in the next stable in 1 month

**Comment by Sen-real** [comment]
> Great thanks.
> That one should be fixed in insiders and should be available in the next stable in 1 month

Cool, thanks!

**Comment by Sen-real** [comment]
@isidorn 
I can't reopen this issue. Should I report a new one?

**Comment by isidorn** [comment]
Nope, already fixed.
